torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -128,6 +129,34 @@ C10_REGISTER_CREATOR(TensorPipeChannelRegistry, cma, makeCmaChannel);
 
 #endif
 
+constexpr static int kNumUvThreads = 16;
```

Comment:
This means we will also need expose channel configs in the future?

Reply:
Indeed, we should have a way to make this configurable. The default I'm putting here, 16, should be enough to cover NICs up to 100 Gbps. On the other hand, it means that whoever is using TensorPipe ends up spawning 16 threads for this channel, even if their NIC is smaller than that. Ideally there should be an auto-sizing mechanism in TensorPipe, or at least a system to turn these on only when needed.

For v1.6, as multiplexing is one of the main benefits of TensorPipe over Gloo, I'm inclined to leave it always on with a large enough number of threads, to ensure everyone can benefit from this. But we definitely need to revisit it.